commit 3e063cc41cb03eada7ed7e56577db8b0c584d9fa
Author: Brian Bennett <bahamat@digitalelf.net>
Date:   Mon Sep 20 15:25:18 2021 -0700

    Better handling around grep/awk

diff --git a/dehydrated b/dehydrated
index d61807e..1e01d2b 100755
--- a/dehydrated
+++ b/dehydrated
@@ -88,7 +88,7 @@ jsonsh() {
   awk_egrep () {
     local pattern_string=$1
 
-    gawk '{
+    awk '{
       while ($0) {
         start=match($0, pattern);
         token=substr($0, start, RLENGTH);
@@ -110,7 +110,7 @@ jsonsh() {
       GREP='egrep -ao'
     fi
 
-    if echo "test string" | egrep -o "test" >/dev/null 2>&1
+    if echo "test string" | egrep -ao "test" >/dev/null 2>&1
     then
       ESCAPE='(\\[^u[:cntrl:]]|\\u[0-9a-fA-F]{4})'
       CHAR='[^[:cntrl:]"\\]'
